avocado/utils/pci.py

```diff
@@ -262,6 +262,22 @@ def get_driver(pci_address):
                 return line.rsplit(None, 1)[-1]
 
 
+def get_slot(pci_address):
+    """
+    Gets the slot number of given PCI address.
+
+    :param pci_address: Any segment of a PCI address
+
+    :return slot number, if not found return none
+    """
+    cmd = "lspci -nnvmms %s" % pci_address
+    output = process.system_output(cmd, ignore_status=True)
+    if output:
+        for line in output.splitlines():
+            if 'PhySlot' in line:
```

Comment:
The problem here is that you accept a partial PCI address to make this search. That can return more than one PCI Slots and you will get the `PhySlot` from the first one which has that key:
```
$ lspci -nnvmms .0
Slot:   00:00.0
Class:  Host bridge [0600]
Vendor: Intel Corporation [8086]
Device: 440FX - 82441FX PMC [Natoma] [1237]
SVendor:        Red Hat, Inc [1af4]
SDevice:        Qemu virtual machine [1100]
Rev:    02

Slot:   00:01.0
Class:  ISA bridge [0601]
Vendor: Intel Corporation [8086]
Device: 82371SB PIIX3 ISA [Natoma/Triton II] [7000]
SVendor:        Red Hat, Inc [1af4]
SDevice:        Qemu virtual machine [1100]

Slot:   00:02.0
Class:  VGA compatible controller [0300]
Vendor: Red Hat, Inc. [1b36]
Device: QXL paravirtual graphic card [0100]
SVendor:        Red Hat, Inc [1af4]
SDevice:        QEMU Virtual Machine [1100]
Rev:    04

Slot:   00:03.0
Class:  Ethernet controller [0200]
Vendor: Red Hat, Inc [1af4]
Device: Virtio network device [1000]
SVendor:        Red Hat, Inc [1af4]
SDevice:        Device [0001]
PhySlot:        3

Slot:   00:04.0
Class:  Communication controller [0780]
Vendor: Red Hat, Inc [1af4]
Device: Virtio console [1003]
SVendor:        Red Hat, Inc [1af4]
SDevice:        Device [0003]
PhySlot:        4

Slot:   00:05.0
Class:  SCSI storage controller [0100]
Vendor: Red Hat, Inc [1af4]
Device: Virtio block device [1001]
SVendor:        Red Hat, Inc [1af4]
SDevice:        Device [0002]
PhySlot:        5

```
I see two ways out:

- Check if the command returns more than one slot and fail when it does.
- Return a pair of PCI full address + PCI slot in a proper structure. Example, searching for `.0` in the machine above, return would be:
```python
[('00:00.0', None), ('00:01.0', None), ('00:02.0', None), ('00:03.0', '3'), ('00:04.0', '4'), ('00:05.0', '5')]
```